stl/inc/xmemory

```diff
@@ -949,21 +949,17 @@ _CONSTEXPR20 void _Destroy_range(_NoThrowFwdIt _First, const _NoThrowSentinel _L
 }
 
 template <class _Size_type>
-_NODISCARD constexpr _Size_type _Convert_size(const size_t _Len) noexcept {
+_NODISCARD constexpr _Size_type _Convert_size(const size_t _Len) noexcept(is_same_v<_Size_type, size_t>) {

```

Comment:
Do you suppose it's worth the trouble to make a "`is_unsigned and no smaller than `size_t`" trait so this also short-circuits for e.g. `uint64_t` on x86?

Reply:
Good question - I doubt it's worth the trouble. Such allocators seem exceptionally rare, and the downside would be very limited (a bit of debug codegen impact; note that this is typically called before performing an allocation so the cost will be near-invisible).